infra/resources.bicep

```
@@ -45,73 +42,53 @@ resource storageAccount 'Microsoft.Storage/storageAccounts@2019-06-01' = {
     minimumTlsVersion: 'TLS1_2'
     allowBlobPublicAccess: false
   }
+
+  resource blobServices 'blobServices' = {
+    name: 'default'
+    resource container 'containers' = {
+      name: functionServiceName
+    }
+  }
 }
 
-resource hostingPlan 'Microsoft.Web/serverfarms@2020-10-01' = {
-  name: '${prefix}-plan'
+resource hostingPlan 'Microsoft.Web/serverfarms@2023-12-01' = {
+  name: '${resourceToken}-plan'
   location: location
   tags: tags
   kind: 'functionapp'
+  sku: {
+    name: 'FC1'
+    tier: 'FlexConsumption'
+  }
   properties: {
     reserved: true
   }
-  sku: {
-    name: 'Y1'
-  }
 }
 
-resource functionApp 'Microsoft.Web/sites@2020-06-01' = {
-  name: '${prefix}-function-app'
-  location: location
-  tags: union(tags, {
-    'azd-service-name': 'api'
-   })
-  kind: 'functionapp,linux'
-  properties: {
-    httpsOnly: true
-    serverFarmId: hostingPlan.id
-    clientAffinityEnabled: false
-    siteConfig: {
-      minTlsVersion: '1.2'
-      linuxFxVersion: 'Python|3.9'
-      appSettings: [
-        {
-           name: 'APPINSIGHTS_INSTRUMENTATIONKEY'
-           value: appInsights.properties.InstrumentationKey
-        }
-        {
-          name: 'AzureWebJobsStorage'
-          value: 'DefaultEndpointsProtocol=https;AccountName=${storageAccount.name};EndpointSuffix=${az.environment().suffixes.storage};AccountKey=${listKeys(storageAccount.id, storageAccount.apiVersion).keys[0].value}'
-        }
-        {
-          name: 'FUNCTIONS_EXTENSION_VERSION'
-          value: '~4'
-        }
-        {
-          name: 'FUNCTIONS_WORKER_RUNTIME'
-          value: 'python'
-        }
-        {
-          name: 'WEBSITE_CONTENTAZUREFILECONNECTIONSTRING'
-          value: 'DefaultEndpointsProtocol=https;AccountName=${storageAccount.name};EndpointSuffix=${az.environment().suffixes.storage};AccountKey=${listKeys(storageAccount.id, storageAccount.apiVersion).keys[0].value}'
-        }
-        {
-          name: 'ENABLE_ORYX_BUILD'
-          value: 'true'
-        }
-        {
-          name: 'SCM_DO_BUILD_DURING_DEPLOYMENT'
-          value: 'true'
-        }
-      ]
+module functionApp 'core/host/functions-flex.bicep' = {
+  name: 'function-app'
+  params: {
+    name: functionServiceName
+    location: location
+    tags: union(tags, { 'azd-service-name': 'api' })
+    alwaysOn: false
+    appSettings: {
+      FUNCTIONS_EXTENSION_VERSION: '~4'
+      AzureWebJobsStorage__accountName: storageAccount.name
```

This is ok but in the future we're recommending using full URIs which will protect you from DNS issues in storage

```suggestion
      AzureWebJobsStorage__blobServiceUri: storageAccount.name
```